### DISCUSSION TITLE
Allow request_charger_current down to 0A

### DISCUSSION THREAD

**Discussion by jurviz** [jhansche/ha-teslafi#40] [Ideas]
I have modified and tested to allow to pause charging by setting the minimum allowed charging to 0.
Would you fix it or make it a feature?
(Modification of line 32 in number.py)

**Comment by jhansche** [comment]
How does this differ from simply toggling the `switch.<car>_charging` switch?
The official Tesla app only allows me to reduce the charger current down to 5A.

I don't currently have this TeslaFi command enabled, so I can't test what happens when I set it below 5 -- what does the app show when it is set below 5? Similarly, what does the app show when it is set to 0?

Given there is already a way to "pause" charging, and the official app's current min charging current, I don't believe this would add any value. But I'm open to other use cases if there's a tangible value-add you can describe?

**Reply by jurviz** [reply]
> How does this differ from simply toggling the switch.<car>_charging switch?<

If charging is not stopped the high voltage relay in the car stays on.
>what does the app show when it is set below 5? Similarly, what does the app show when it is set to 0?

I don't know what the app shows, but i know that the charge current is reduced below 5A down to 0A.
I'm using it to pause the charging when my solar panels don't produce enough power for the car and the house combined.
My charger only allows setting it down to 6A .

**Reply by jhansche** [reply]
I enabled the command, and at first it looked like dropping it to 1A actually didn't work, because the Tesla app still showed 5A. But I tried it again, and I do see that it appears to work correctly down to 1A. But I can reproduce fairly consistently, that it often does not respect a value below 5A at first.. seems to take a few attempts before it accepts it.

However, setting it to 0A did not work correctly at all, no matter how many times I tested it. It continued to report 1A, not 0A.

![Screenshot_20240608-223837](https://github.com/jhansche/ha-teslafi/assets/99272/19f92543-6c4b-4cf9-a425-15c657bbcfeb)

I also continue to see 1A when I go to https://teslafi.com/index.php after setting it from the API.

So I'm not convinced that this actually does anything. What evidence are you seeing that indicates that 0A is actually accepted and working?

If it is not able to accept 0A, then allowing it as an option would be misleading at best.

If you want your car to stop charging, then I think the proper solution is the charging switch. Can you explain why you care about the high voltage relay remaining closed, if your goal is actually that you don't want the car to be drawing power? It sounds like you're implying there is a distinction between setting current to 1A, and setting current to 0A, but my own tests show no distinction between those, which means it's actually not drawing 0W, but 120W (or 240W if 240V), in both cases.

**Reply by jurviz** [reply]
> So I'm not convinced that this actually does anything. What evidence are you seeing that indicates that 0A is actually accepted and working?

I don't have the evidence right now. My solar panels are in Spain and i'm in Germany right now. I will continue the tests in July when i'm back in Spain. I can see what the current draw of the charger is at that time.
I will also test what is happening when stopping the charge instead of setting charging amp to 0A. There was a discussion in the german Tesla forum that stopping charging repeatedly would increase the wear and tear of the high voltage relay.

**Reply by jurviz** [reply]
The history of the charger in Home Assistant shows that the power drawn by the car drops to 0.2A when setting setting the charging amps to 0A. which is 46W for 230V.

**Comment by jurviz** [comment]
Today i have seen that SoC drops when setting the charging power to 0A in the car. It looks like the car needs more than 0.2A from the charger and takes the rest from the battery. So it makes sense to keep the 1A lower boundary.
